evalml/pipelines/time_series_classification_pipelines.py

```diff
@@ -261,6 +260,7 @@ def predict_in_sample(self, X, y, X_train, y_train, objective=None):
             proba = proba.iloc[:, 1]
             if objective is None:
                 predictions = proba > self.threshold
+                predictions = predictions.astype(int)
```

Comment:
This is necessary because it returns bools, which we don't know how to map back (since we expect ints!). Although we treated bools and 0/1 as interchangeable, it actually matters now that we're storing a mapping.

Previously with sklearn this wasn't an issue because they used `np.searchsorted` which would treat ints and bools the same. This is no longer true. :)